### PR TITLE
Improve CI check for Analyzer

### PR DESCRIPTION
**Pull request by novikd**
### Changelog category (leave one):

- Build/Testing/Packaging Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Improve CI check with an enabled analyzer. Now it should be green if only tests from `tests/broken_tests.txt` are broken.

### CODE REVIEW

File: docker/test/util/process_functional_tests_result.py [hunk 1]
```diff
@@ -59,8 +59,12 @@ def process_test_log(log_path, broken_tests):
 
                 total += 1
                 if TIMEOUT_SIGN in line:
-                    failed += 1
-                    test_results.append((test_name, "Timeout", test_time, []))
+                    if test_name in broken_tests:
+                        success += 1
+                        test_results.append((test_name, "OK", test_time, []))
```
Comment (davenger) [inline docker/test/util/process_functional_tests_result.py:64]: Is it possible to write smth like "KNOWN_BROKEN" instead of "OK" here?

**Comment by robot-clickhouse-ci-1** [comment]
This is an automated comment for commit b13e3abc4d77fcd2bcc3069d90bb4481e6fca0d1 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/49562/b13e3abc4d77fcd2bcc3069d90bb4481e6fca0d1/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

**Review by vdimir — APPROVED** [review]

File: docker/test/util/process_functional_tests_result.py (hunk 1 above)
Comment (vdimir) [reply docker/test/util/process_functional_tests_result.py:64]: Status can be wrong if we calculate the number of `OK`, isn't it?

Comment (Felixoid) [reply docker/test/util/process_functional_tests_result.py:64]: It rather affects the color of the report. Anything unknown is written as yellow color

Comment (Felixoid) [reply docker/test/util/process_functional_tests_result.py]: https://github.com/ClickHouse/ClickHouse/blob/master/tests/ci/report.py#L283

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**